wolfssl/internal.h

```diff
@@ -5084,6 +5123,46 @@ WOLFSSL_LOCAL int oid2nid(word32 oid, int grp);
 WOLFSSL_LOCAL word32 nid2oid(int nid, int grp);
 #endif
 
+#ifdef WOLFSSL_DTLS13
+
+WOLFSSL_LOCAL void dtls13_do_legacy_version(WOLFSSL *ssl, ProtocolVersion *pv,
+                                            int *wantDowngrade);
+
+WOLFSSL_LOCAL int dtls13_derive_sn_keys(WOLFSSL *ssl, int provision);
+WOLFSSL_LOCAL int dtls13_add_headers(byte *output, word32 length,
+                                     enum HandShakeType hs_type, WOLFSSL *ssl);
+WOLFSSL_LOCAL int dtls13_get_headers_length(WOLFSSL *ssl, enum HandShakeType type);
+WOLFSSL_LOCAL int dtls13_get_rl_header_length(WOLFSSL *ssl, int is_encrypted);
+WOLFSSL_LOCAL int dtls13_rl_add_ciphertext_header(WOLFSSL *ssl, byte *out, size_t length);
+WOLFSSL_LOCAL int dtls13_encrypt_record_number(WOLFSSL *ssl, byte *hdr);
+WOLFSSL_LOCAL int dtls13_is_unified_header(byte header_flags);
+WOLFSSL_LOCAL int dtls13_parse_unifed_record_layer(WOLFSSL *ssl,
+                                                   const byte *input,
```

Comment:
This makes me think you have too many parameters.

Reply:
yes I agree, I'll make a struct to hold the parameters